tools/ddrgen/src/scanners/dwarf/DwarfScanner.cpp

```diff
@@ -77,9 +77,11 @@ DwarfScanner::getBlacklist(Dwarf_Die die)
 		goto Failed;
 	}
 
-	if (DW_DLV_ERROR == dwarf_formstring(attr, &compDir, &error)) {
-		ERRMSG("Getting compilation directory string: %s\n", dwarf_errmsg(error));
-		goto Failed;
+	if (NULL != attr) {
```

Comment:
If the attribute DW_AT_comp_dir does not exist, the Dwarf API will not necessarily just return null. I've had it often crash when attempting to use dwarf_attr() on an attribute that a Die does not have. Thus, your change can still segfault. The dwarf spec states one should use dwarf_hasattr().

The correct way to do this with the dwarf_hasattr() function is like this:
Dwarf_Bool hasAttr = false;
if (DW_DLV_ERROR == dwarf_hasattr(die, DW_AT_comp_dir, &hasAttr, &err)) {
   ERRMSG(...
   rc = DDR_RC_ERROR;
   goto Failed;
}
if (hasAttr) {
    if (DW_DLV_ERROR == dwarf_attr(die, DW_AT_comp_dir, &attr, &error)) {
        ERRMSG("Getting compilation directory attribute: %s\n", dwarf_errmsg(error));
        goto Failed;
    }
    ....